src/services/formatting/smartIndenter.ts

```diff
@@ -464,9 +471,29 @@ namespace ts.formatting {
                 case SyntaxKind.GetAccessor:
                 case SyntaxKind.SetAccessor:
                     return child !== SyntaxKind.Block;
+                case SyntaxKind.ExportDeclaration:
+                    return child !== SyntaxKind.NamedExports;
                 default:
                     return false;
             }
         }
+
+        function hasBraceShell(node: Node) {

```

Comment:
`hasBraceShell` function is not to check completeness as `ImportClause` node can omit its optional braces. I think `isCompletedNode` function is to check a node has its required children.